applications/StructuralMechanicsApplication/custom_elements/truss_element_3D2N.cpp

```diff
@@ -634,22 +634,58 @@ void TrussElement3D2N::AddExplicitContribution(
     Variable<array_1d<double, 3>> &rDestinationVariable,
     const ProcessInfo &rCurrentProcessInfo) {
   KRATOS_TRY;
-  if (rRHSVariable == RESIDUAL_VECTOR &&
-      rDestinationVariable == FORCE_RESIDUAL) {
 
-    for (int i = 0; i < msNumberOfNodes; ++i) {
-      int index = msDimension * i;
+  bounded_vector<double, msLocalSize> damping_residual_contribution =
+      ZeroVector(msLocalSize);
+  // calculate damping contribution to residual -->
+  if ((this->GetProperties().Has(RAYLEIGH_ALPHA) ||
+       this->GetProperties().Has(RAYLEIGH_BETA)) &&
+      (rDestinationVariable != NODAL_INERTIA)) {
+    Vector current_nodal_velocities = ZeroVector(msLocalSize);
+    this->GetFirstDerivativesVector(current_nodal_velocities);
+    Matrix damping_matrix = ZeroMatrix(msLocalSize, msLocalSize);
+    ProcessInfo temp_process_information; // cant pass const ProcessInfo
+    this->CalculateDampingMatrix(damping_matrix, temp_process_information);
+    // current residual contribution due to damping
+    damping_residual_contribution =
+        prod(damping_matrix, current_nodal_velocities);
+  }
 
-      GetGeometry()[i].SetLock();
+  if (rRHSVariable == RESIDUAL_VECTOR &&
+      rDestinationVariable == FORCE_RESIDUAL) {
 
+    for (size_t i = 0; i < msNumberOfNodes; ++i) {
+      size_t index = msDimension * i;
       array_1d<double, 3> &r_force_residual =
           GetGeometry()[i].FastGetSolutionStepValue(FORCE_RESIDUAL);
-
-      for (int j = 0; j < msDimension; ++j) {
-        r_force_residual[j] += rRHSVector[index + j];
+      for (size_t j = 0; j < msDimension; ++j) {
+#pragma omp atomic
+        r_force_residual[j] +=
+            rRHSVector[index + j] - damping_residual_contribution[index + j];
       }
+    }
+  }
+
+  if (rDestinationVariable == NODAL_INERTIA) {
+
+    Matrix element_mass_matrix = ZeroMatrix(msLocalSize, msLocalSize);
+    ProcessInfo temp_info; // Dummy

```

Comment:
uff, this is a very heavy object. I would not create an empty one here...

Reply:
the problem is that the virtual CalculateMassMatrix needs a non const ProcessInfo but the virtual addexplicitcontr needs a const ProcessInfo